### PR TITLE
Use webpack and node-fetch to reduce package size

### PR DESCRIPTION
**Pull request by jetersen**
Using webpack and managing our dependencies we can greatly reduce our package size.

Reduced package size from 245KB to 29KB
Before adding webpack the package size was 1.93MB

Before 4bcd7b3 reduce package size by using node-fetch
```
Hash: 6e7f71f3817d2bdc2b3a
Version: webpack 4.41.5
Time: 3761ms
Built at: 12.01.2020 18.40.00
           Asset      Size  Chunks                   Chunk Names
    extension.js   816 KiB       0  [emitted]        main
extension.js.map  1.88 MiB       0  [emitted] [dev]  main
Entrypoint main = extension.js extension.js.map
  [3] external "util" 42 bytes {0} [built]
  [4] external "crypto" 42 bytes {0} [built]
 [10] external "url" 42 bytes {0} [built]
 [17] external "stream" 42 bytes {0} [built]
 [20] external "http" 42 bytes {0} [built]
 [29] external "net" 42 bytes {0} [built]
 [30] external "https" 42 bytes {0} [built]
 [36] external "punycode" 42 bytes {0} [built]
 [46] external "fs" 42 bytes {0} [built]
 [51] external "async_hooks" 42 bytes {0} [built]
 [98] ./src/extension.ts 2.27 KiB {0} [built]
 [99] external "vscode" 42 bytes {0} [built]
[101] (webpack)/buildin/module.js 497 bytes {0} [built]
[152] external "zlib" 42 bytes {0} [built]
[258] external "os" 42 bytes {0} [built]
    + 244 hidden modules
 DONE  Packaged: C:\git\code\jcasc-vscode-extension\jcasc-plugin-0.0.1.vsix (7 files, 244.78KB)
```

After 4bcd7b3 reduce package size by using node-fetch
```
Hash: 324d2b47e3e2302bc005
Version: webpack 4.41.5
Time: 1955ms
Built at: 12.01.2020 18.41.38
           Asset      Size  Chunks                   Chunk Names
    extension.js  19.7 KiB       0  [emitted]        main
extension.js.map  73.6 KiB       0  [emitted] [dev]  main
Entrypoint main = extension.js extension.js.map
 [0] external "stream" 42 bytes {0} [built]
 [1] external "zlib" 42 bytes {0} [built]
 [2] external "url" 42 bytes {0} [built]
 [3] external "http" 42 bytes {0} [built]
 [4] external "https" 42 bytes {0} [built]
 [5] ./src/extension.ts 3.41 KiB {0} [built]
 [6] external "vscode" 42 bytes {0} [built]
 [8] external "util" 42 bytes {0} [built]
 [9] external "fs" 42 bytes {0} [built]
[10] external "path" 42 bytes {0} [built]
    + 1 hidden module
 DONE  Packaged: C:\git\code\jcasc-vscode-extension\jcasc-plugin-0.0.1.vsix (7 files, 29.39KB)
```

### CODE REVIEW

**Comment by jetersen** [comment]
I cannot view the azure pipeline?

Perhaps switching to GitHub actions is a better option?

File: src/extension.ts [hunk 1]
```diff
@@ -1,9 +1,11 @@
 // The module 'vscode' contains the VS Code extensibility API
 // Import the module and reference it with the alias vscode in your code below
 import * as vscode from 'vscode';
-
-const rp = require("request-promise");
-var fs = require("fs");
+import fetch from 'node-fetch';
+import { promisify } from 'util';
+import { pipeline } from 'stream';
+import { createWriteStream } from "fs";
```
Comment (jetersen) [inline src/extension.ts]: I need prettier 😭
```suggestion
import { createWriteStream } from 'fs';
```

**Comment by timja** [comment]
> I cannot view the azure pipeline?

You should be able to now, seem like a bug =/, it's configured for public as far as I can tell

> Perhaps switching to GitHub actions is a better option?

Possibly, we just used AzureDevOps because there was a documented pipeline in the vscode docs that we could use for release and it's required that you have an AzureDevOps account for publishing.

File: Jenkinsfile [hunk 2]
```diff
@@ -3,19 +3,14 @@ pipeline {
     stages {
         stage('build') {
             steps {
-                 sh '''
-                    npm install -g typescript
-                    npm install
-                    npm run vscode:prepublish
-                    '''
+                sh 'npm install'
```
Comment (timja) [inline Jenkinsfile]: the Jenkinsfile isn't used, not sure if it's worth keeping?

Comment (jetersen) [reply Jenkinsfile]: Let me remove it in a separate PR then :)
Same goes for travis

**Review by timja — APPROVED** [review]

**Pull request merged**